p2p/cancel_token.py

```diff
@@ -70,17 +70,19 @@ def __str__(self):
         return
 
 
-async def wait_with_token(future: Awaitable,
+async def wait_with_token(futures: List[Awaitable],
```

Comment:
Looking at the way this is used, I'm wondering if the UX for using this would be better if `futures` was done as `*futures`.

```
# current
wait_with_token([the_future], cancel_token)

# positional
wait_with_token(the_future, cancel_token=cancel_token)

# positional and change `cancel_token` to `token`
wait_with_token(the_future, token=cancel_token)
```

It's not shorter, but it does avoid the awkward need to wrap the first argument in an iterable.

Reply:
Yeah, that's a nice improvement